pylint/reporters/reports_handler_mix_in.py

```diff
@@ -33,45 +39,55 @@ def register_report(self, reportid, r_title, r_cb, checker):
         reportid = reportid.upper()
         self._reports[checker].append((reportid, r_title, r_cb))
 
-    def enable_report(self, reportid):
+    def enable_report(self, reportid: str) -> None:
         """disable the report of the given id"""
         reportid = reportid.upper()
         self._reports_state[reportid] = True
 
-    def disable_report(self, reportid):
+    def disable_report(self, reportid: str) -> None:
         """disable the report of the given id"""
         reportid = reportid.upper()
         self._reports_state[reportid] = False
 
-    def report_is_enabled(self, reportid):
+    def report_is_enabled(self, reportid: str) -> bool:
         """return true if the report associated to the given identifier is
         enabled
         """
         return self._reports_state.get(reportid, True)
 
-    def make_reports(self, stats, old_stats):
+    def make_reports(
+        self: Union[BaseChecker, "ReportsHandlerMixIn"],
+        stats: Dict[str, Any],
+        old_stats: Dict,
+    ) -> Section:
         """render registered reports"""
-        sect = Section("Report", f"{self.stats['statement']} statements analysed.")
-        for checker in self.report_order():
-            for reportid, r_title, r_cb in self._reports[checker]:
-                if not self.report_is_enabled(reportid):
-                    continue
-                report_sect = Section(r_title)
-                try:
-                    r_cb(report_sect, stats, old_stats)
-                except EmptyReportError:
-                    continue
-                report_sect.report_id = reportid
-                sect.append(report_sect)
-        return sect
+        if isinstance(self, BaseChecker) and isinstance(self, ReportsHandlerMixIn):
```

Comment:
```suggestion
        if not isinstance(self, BaseChecker) or not isinstance(self, ReportsHandlerMixIn):
            raise TypeError
```
Premature return so the following code is simpler to understand with less to keep in mind and less indented.